### PR TITLE
Adding normalization support in FeatureExtractionPipeline

### LINKED ISSUE #20: Support normalization like in PyTorch
When we run `NewFeatureExtractionPipeline` with model `sentence-transformers/all-mpnet-base-v2` the result is not normalized. Please refer to https://huggingface.co/sentence-transformers/all-mpnet-base-v2
What we get from pipeline is exactly like `sentence_embeddings` after `mean_pooling` and then need to run `F.normalize` - only after normalization the embeddings are equivalent to result of `encode()` from `SentenceTransformer` python package.
I searched myself for implementation of such normalization in Go packages and unfortunately didn't find any, the most closest stuff was in https://github.com/sugarme/gotch with their rich tensor functionality (but still `normalize` function is missing there). So I implemented it myself based on algorithm provided here:
https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#normalize
The code is not long but required a lot of reverse engineering and debugging, so I will be glad to save time of other developers.
Please let me know if you want to add this functionality, either as new `normalize` function, or directly as output from `NewFeatureExtractionPipeline` (maybe with boolean param if to normalize).

### PR DESCRIPTION
**Pull request by sedletsky-f5**
Fixes #20

Tested with the following code and `go run -tags GOTCH`:
```
func main() {
	session, err := hugot.NewSession()
	defer func(session *hugot.Session) {
		err := session.Destroy()
		check(err)
	}(session)
	pipeline, err := session.NewFeatureExtractionPipeline("sentence-transformers_all-mpnet-base-v2", "testPipeline", pipelines.WithNormalization())
	check(err)
	prompts := []string{
		...
	}
	batchResult, err := pipeline.RunPipeline(prompts)
	check(err)
	fmt.Printf("%+v\n", batchResult)
}
```
Works as expected, like externally implemented `normalize()` function.

In case somebody wants to use this support, need to follow complicated installation steps for `gotch` package in https://github.com/sugarme/gotch/blob/master/README.md
To avoid overhead for others, it is done with build flag and by default not enabled - so `Dockerfile` should not be extended with all those steps and pipeline should pass as in past.

Pipeline passed: https://github.com/knights-analytics/hugot/actions/runs/8864746623

### CODE REVIEW

File: hugot.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ type TokenClassificationOption = pipelines.PipelineOption[*pipelines.TokenClassi
 // TextClassificationOption is an option for a text classification pipeline
 type TextClassificationOption = pipelines.PipelineOption[*pipelines.TextClassificationPipeline]
 
+// FeatureExtractionOption is an option for a text classification pipeline
```
Comment (sedletsky-f5) [inline hugot.go:57]: typo in comment: text classification => feature extraction
